dubbo-serialization/dubbo-serialization-msgpack/src/main/java/org/apache/dubbo/common/serialize/msgpack/MsgpackObjectInput.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.common.serialize.msgpack;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.dubbo.common.serialize.ObjectInput;
+import org.apache.dubbo.common.utils.PojoUtils;
+
+import org.msgpack.jackson.dataformat.MessagePackFactory;
+
+import java.io.*;
+import java.lang.reflect.Type;
+
+public class MsgpackObjectInput implements ObjectInput {
+
+    private final InputStream in;
+
+    private ObjectMapper om;
+
+    public MsgpackObjectInput(InputStream in) {
+        this.in = in;
+        om = new ObjectMapper(new MessagePackFactory());
+        om.configure(JsonParser.Feature.AUTO_CLOSE_SOURCE, false);
```

Comment:
Why can't we initialize `ObjectMapper` in the same way in both `MsgpackObjectInput` and `MsgpackObjectOutput` class?
```java
    public MsgpackObjectOutput(OutputStream out) {
        this.out = out;
        om = new ObjectMapper(new MessagePackFactory());
    }
```

Reply:
get